eth/core/state/journal/transient_storage.go

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: BUSL-1.1
+//
+// Copyright (C) 2023, Berachain Foundation. All rights reserved.
+// Use of this software is govered by the Business Source License included
+// in the LICENSE file of this repository and at www.mariadb.com/bsl11.
+//
+// ANY USE OF THE LICENSED WORK IN VIOLATION OF THIS LICENSE WILL AUTOMATICALLY
+// TERMINATE YOUR RIGHTS UNDER THIS LICENSE FOR THE CURRENT AND ALL OTHER
+// VERSIONS OF THE LICENSED WORK.
+//
+// THIS LICENSE DOES NOT GRANT YOU ANY RIGHT IN ANY TRADEMARK OR LOGO OF
+// LICENSOR OR ITS AFFILIATES (PROVIDED THAT YOU MAY USE A TRADEMARK OR LOGO OF
+// LICENSOR AS EXPRESSLY REQUIRED BY THIS LICENSE).
+//
+// TO THE EXTENT PERMITTED BY APPLICABLE LAW, THE LICENSED WORK IS PROVIDED ON
+// AN “AS IS” BASIS. LICENSOR HEREBY DISCLAIMS ALL WARRANTIES AND CONDITIONS,
+// EXPRESS OR IMPLIED, INCLUDING (WITHOUT LIMITATION) WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
+// TITLE.
+
+package journal
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"pkg.berachain.dev/polaris/lib/ds"
+	"pkg.berachain.dev/polaris/lib/ds/stack"
+)
+
+// transientState is a representation of EIP-1153 "Transient Storage".
+type transientState map[common.Address]map[common.Hash]common.Hash
+
+// Set sets the transient storage state `value` for `key` at the given `addr`.
+func (t transientState) Set(addr common.Address, key, value common.Hash) {
+	if _, ok := t[addr]; !ok {
+		t[addr] = make(map[common.Hash]common.Hash)
+	}
+	t[addr][key] = value
+}
+
+// Get gets the transient storage state for `key` at the given `addr`.
+func (t transientState) Get(addr common.Address, key common.Hash) common.Hash {
+	val, ok := t[addr]
+	if !ok {
+		return common.Hash{}
+	}
+	hash, ok := val[key]
+	if !ok {
+		return common.Hash{}
+	}
+	return hash
```

Comment:
lmaooo mb here. you can revert these changes and just return `val[key]`. I realized go maps with a value of an array (in this case `common.Hash`) will actually return the empty array and not nil. nil is only returned when the value is a pointer/interface.
https://go.dev/play/p/IqH6gRQn9sY

Reply:
its okay, i also missed that lmao